### PR TITLE
Change log level in ChainReplicationProtocol for recover and propagate

### PR DESCRIPTION
**Pull request by StrikingThirteen**
## Overview

Description:
Changing the log level of recover and propagate in ChainReplicationProtocol to decrease the amount of logging.
Also tightening the visibility of recover and propagate.

Why should this be merged: 
Too many logs being emitted 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/runtime/view/replication/ChainReplicationProtocol.java [hunk 1]
```diff
@@ -143,13 +143,14 @@ public ILogData peek(RuntimeLayout runtimeLayout, long globalAddress) {
...
      */
-    protected void propagate(RuntimeLayout runtimeLayout,
+    private void propagate(RuntimeLayout runtimeLayout,
                              long globalAddress,
                              @Nullable ILogData data) {
         int numUnits = runtimeLayout.getLayout().getSegmentLength(globalAddress);
 
         for (int i = 1; i < numUnits; i++) {
-            log.info("Propogate[{}]: chain {}/{}", Token.of(runtimeLayout.getLayout().getEpoch(), globalAddress),
+            log.trace("Propogate[{}]: chain {}/{}", Token.of(runtimeLayout.getLayout().getEpoch(),
```
Comment (annym) [inline runtime/src/main/java/org/corfudb/runtime/view/replication/ChainReplicationProtocol.java]: Now that we're on this, can you please correct the typo in 'propagate' here and also in the catch clause..

Comment (StrikingThirteen) [reply runtime/src/main/java/org/corfudb/runtime/view/replication/ChainReplicationProtocol.java]: Done

File: runtime/src/main/java/org/corfudb/runtime/view/replication/ChainReplicationProtocol.java [hunk 2]
```diff
@@ -166,7 +167,7 @@ protected void propagate(RuntimeLayout runtimeLayout,
                             .fillHole(token), OverwriteException.class);
                 }
             } catch (OverwriteException oe) {
-                log.trace("Propogate[{}]: Completed by other writer", globalAddress);
+                log.trace("Propagate[{}]: Completed by other writer", globalAddress);
```
Comment (Maithem) [inline runtime/src/main/java/org/corfudb/runtime/view/replication/ChainReplicationProtocol.java]: This is a rare event. I think we should make it INFO

Comment (annym) [reply runtime/src/main/java/org/corfudb/runtime/view/replication/ChainReplicationProtocol.java]: I agree with this, this should be at info level.

**Pull request merged**